sdks/go/pkg/beam/core/runtime/exec/translate.go

```diff
@@ -495,10 +495,22 @@ func (b *builder) makeLink(from string, id linkID) (Node, error) {
 							} else if ms := spec.GetMapSpec(); ms != nil {
 								cID = ms.ValueCoderId
 								kcID = ms.KeyCoderId
+							} else if ss := spec.GetSetSpec(); ss != nil {
+								kcID = ss.ElementCoderId
+							} else {
+								a := spec.GetSpec().(*pipepb.StateSpec_SetSpec)
```

Comment:
Is it possible to directly use `spec.GetSpec()` in error statement below instead of checking for `SetSpec` value here?